plugin/trino-hive/src/test/java/io/trino/plugin/hive/coercions/TestDateCoercer.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.coercions;
+
+import io.trino.spi.type.DateType;
+import io.trino.spi.type.VarcharType;
+import org.testng.annotations.Test;
+
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.trino.plugin.hive.HiveTimestampPrecision.NANOSECONDS;
+import static io.trino.plugin.hive.coercions.TestTimestampCoercer.assertCoercions;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestDateCoercer
+{
+    @Test
+    public void testValidUnboundedVarcharToDate()
+    {
+        assertCoercions(VarcharType.createUnboundedVarcharType(), utf8Slice("2023-04-13"), DateType.DATE, 19460L, NANOSECONDS);
+    }
+
+    @Test
+    public void testValidBounded10VarcharToDate()
+    {
+        assertCoercions(VarcharType.createVarcharType(10), utf8Slice("2023-04-13"), DateType.DATE, 19460L, NANOSECONDS);
+    }
+
+    @Test
+    public void testReturnsNullWhenStringIsInvalidDate()
+    {
+        assertCoercions(VarcharType.createUnboundedVarcharType(), utf8Slice("notSoValidDate"), DateType.DATE, null, NANOSECONDS);
+    }
+
+    @Test
+    public void testThrowsWhenDateHasValidFormatButInvalidValues()
+    {
+        assertThatThrownBy(() -> assertCoercions(VarcharType.createUnboundedVarcharType(), utf8Slice("2023-04-35"), DateType.DATE, null, NANOSECONDS))
+                .hasMessageMatching(".*Invalid date value: .*");
```

Comment:
You could add as well the US way of specifying dates yyyy-dd-MM

```
        assertThatThrownBy(() -> assertCoercions(VarcharType.createUnboundedVarcharType(), utf8Slice("2023-30-04"), DateType.DATE, null, NANOSECONDS))
                .hasMessageMatching(".*Invalid date value: .*");

```

Reply:
i need to check if hive supports this

Reply:
It does not, it treats it as an overflow

Reply:
Yes, i was actually suggesting as well that an error should be thrown / null fallback should happen in such cases.